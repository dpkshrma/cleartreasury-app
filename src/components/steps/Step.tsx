import React from "react";
import { CheckCircleIcon } from "@heroicons/react/solid";

export const State = {
  ACTIVE: "cursor-default border-t-4 border-green-500 bg-white",
  COMPLETE: "cursor-pointer text-gray-400 border-t-4 border-gray-400",
  DEFAULT: "cursor-default border-t-4 border-transparent",
} as const;

type Values<T> = T[keyof T];

export interface StepProps {
  step?: number;
  title?: string;
  state?: Values<typeof State>;
  isEnabled?: boolean;
  onClick?(): void;
}

const Step = ({
  step,
  title,
  isEnabled,
  onClick,
  state = State.DEFAULT,
}: StepProps): JSX.Element => {
  const handleOnClick = () => {
    if (!isEnabled) return;
    onClick();
  };

  return (
    <div
      className={`flex flex-1 pb-4 pt-3 px-2 ${state}`}
      onClick={handleOnClick}
    >
      {state === State.COMPLETE ? (
        CheckCircleIcon({ className: "w-7 h-7 flex-shrink-0 font-normal" })
      ) : (
        <span className="flex justify-center p-0.5 w-6 h-6 rounded-2xl text-xs text-white bg-teal-700">
          {step + 1}
        </span>
      )}

      <span className="ml-2">{title}</span>
    </div>
  );
};

export default Step;
