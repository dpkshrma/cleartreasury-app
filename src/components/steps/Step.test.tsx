import * as React from "react";
import { render } from "../../../test/testUtils";
import * as Stories from "./Step.stories";

describe("Step", () => {
  it("matches snapshot for default state", () => {
    const { asFragment } = render(
      <Stories.Default {...Stories.Default.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for enabled state", () => {
    const { asFragment } = render(
      <Stories.Enabled {...Stories.Enabled.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for active state", () => {
    const { asFragment } = render(<Stories.Active {...Stories.Active.args} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for complete state", () => {
    const { asFragment } = render(
      <Stories.Complete {...Stories.Complete.args} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
