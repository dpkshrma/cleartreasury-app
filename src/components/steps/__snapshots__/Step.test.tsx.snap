// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Step matches snapshot for active state 1`] = `
<DocumentFragment>
  <div
    class="group flex flex-1 px-2 py-1 md:pb-4 md:pt-3 text-sm lg:text-base md:leading-tight border-l-4 sm:border-l-0 sm:border-t-4 max-w-xs cursor-default text-gray-600 border-green-600 sm:bg-theme-color-surface"
  >
    <span
      class="flex flex-shrink-0 justify-center w-6 h-6 rounded-2xl text-xs text-white pt-0.5 bg-teal-700"
    >
      1
    </span>
    <span
      class="ml-2"
    >
      Active
    </span>
  </div>
</DocumentFragment>
`;

exports[`Step matches snapshot for complete state 1`] = `
<DocumentFragment>
  <div
    class="group flex flex-1 px-2 py-1 md:pb-4 md:pt-3 text-sm lg:text-base md:leading-tight border-l-4 sm:border-l-0 sm:border-t-4 max-w-xs cursor-pointer text-gray-400 border-gray-400 transition hover:text-teal-700 hover:border-teal-700"
  >
    <span
      class="flex flex-shrink-0 justify-center w-6 h-6 rounded-2xl text-xs text-white items-center bg-gray-400 transition group-hover:bg-teal-700"
    >
      <svg
        class="w-3 h-3 flex-shrink-0"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <span
      class="ml-2"
    >
      Complete
    </span>
  </div>
</DocumentFragment>
`;

exports[`Step matches snapshot for default state 1`] = `
<DocumentFragment>
  <div
    class="group flex flex-1 px-2 py-1 md:pb-4 md:pt-3 text-sm lg:text-base md:leading-tight border-l-4 sm:border-l-0 sm:border-t-4 max-w-xs cursor-default text-gray-600 border-transparent"
  >
    <span
      class="flex flex-shrink-0 justify-center w-6 h-6 rounded-2xl text-xs text-white pt-0.5 bg-teal-700"
    >
      1
    </span>
    <span
      class="ml-2"
    >
      Default
    </span>
  </div>
</DocumentFragment>
`;

exports[`Step matches snapshot for enabled state 1`] = `
<DocumentFragment>
  <div
    class="group flex flex-1 px-2 py-1 md:pb-4 md:pt-3 text-sm lg:text-base md:leading-tight border-l-4 sm:border-l-0 sm:border-t-4 max-w-xs cursor-default text-gray-600 border-transparent"
  >
    <span
      class="flex flex-shrink-0 justify-center w-6 h-6 rounded-2xl text-xs text-white pt-0.5 bg-teal-700"
    >
      1
    </span>
    <span
      class="ml-2"
    >
      Default
    </span>
  </div>
</DocumentFragment>
`;
